arcaflow_plugin_pcp/pcp_plugin.py

```diff
@@ -145,81 +145,79 @@ def start_pcp(
+                        f"{pcp2json_return[1]} "
                         f"Retrying in {params.pmlogger_interval} seconds..."
                     )
                     sleep(params.pmlogger_interval)
                     continue
-                # After max_retries, if there is still an exception in
-                # the pcp2json command, we return an error.
                 else:
-                    return "error", Error(
-                        "{} failed with return code {}:\n{}".format(
-                            error.cmd[0], error.returncode, error.output
-                        )
+                    pcp_out_json = json.loads(
+                        pcp2json_return[1].strip().split("\n", 2)[2]
                     )
+                    pcp_metrics_list = pcp_out_json["@pcp"]["@hosts"][0]["@metrics"]
 
             if params.generate_csv:
-                # Initialize (optional) pcp2csv command
-                pcp2csv_cmd = [
-                    "/usr/bin/pcp2csv",
-                ]
-
-                pcp2csv_cmd.extend(pcp2_flags)
-                pcp2csv_cmd.extend(metrics)
-
-                try:
-                    csv_out = subprocess.check_output(
-                        pcp2csv_cmd,
-                        text=True,
-                        stderr=subprocess.STDOUT,
-                    )
-                    print(csv_out)
-                except subprocess.CalledProcessError as error:
-                    print(
-                        "{} failed with return code {}:\n{}".format(
-                            error.cmd[0], error.returncode, error.output
-                        )
-                    )
+                # Send the CSV format to stdout, if requested
+                print(pcp2csv_return[1])
 
-            # If pcp2json completes without an exception, we return success.
-            pcp_metrics_list = pcp_out_json["@pcp"]["@hosts"][0]["@metrics"]
+            # If pcp2json or pcp2csv completes without an exception, we return success.
             return "success", PerfOutput(pcp_metrics_list)
 
-        # Since the above while loop should always return either success or error,
-        # and should never come to its natural end, if we get here, something
-        # unexpected went wrong.
-        return "error", Error("Unknown failure attempting to process pmlogger output")
+        else:
+            # Return the appropriate error condition after max_retries
+            if "error" in pcp2json_return[0]:
+                return pcp2json_return
+            elif "error" in pcp2csv_return[0]:
+                return pcp2csv_return
+            else:
+                # Since the above for loop should always return either success or error,
+                # and should never come to its natural end, if we get here, something
+                # unexpected went wrong.
+                return "error", Error(
+                    "Unknown failure attempting to process pmlogger output"
+                )
```

Comment:
Since there is no `break` statement in the `for` loop, this `else` clause will always be executed (unless control reaches the `return` statement at line 206), so you might as well get rid of the `else` and unindent these lines.